charts/openshift-etcd-backup/templates/pvc.yaml

```
@@ -0,0 +1,17 @@
+{{- if .Values.persistence.provisioning.enabled }}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ include "openshift-etcd-backup.fullname" . }}
+  labels:
+    {{- include "openshift-etcd-backup.labels" . | nindent 4 }}
+spec:
+  {{- if .Values.persistence.provisioning.enabled }}
```

The wrong value has been used in the if.

```suggestion
  {{- if .Values.persistence.provisioning.storageClass }}
```

---

No. I guess what @tongpu means here is, if the StorageClass is set in provisioning, then add it to the pvc.yaml, but if not, keep it empty and it will take the default storageclass.

---

but that's what i do right?!

---

not exactly, you're setting a storageClassName even if no storageClass is defined in `provisioning.storageClass`

---

If I set provisoning.enabled to true, but keep the storageClass empty, it would resutlt in adding `storageClassName: ""` would that work? I'm not sure.

---

yes it does work, if the `storageClass` is empty, it uses the default one

---

but i will test and clean that out as well, to be more consistent

---

Just tried it out...

The first one is created with `StorageClassName: ""` while the second one has no `StorageClassName` at all in `spec:`

```
NAME   STATUS    VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   AGE
tmp    Pending                                                     27s
tmp2   Pending                                      default        2s
```

---

When not specifying the `storageClassName` parameter you either get not storage at all, because no default Storage Class has been defined, or the storage would be created using the default Storage Class: https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1
The behavior of using `storageClassName: ""` is different and should never provision storage using the default Storage Class.